### PR TITLE
[BOLT][DWARF][NFC] Replace usages of GDBIndex functions

### PR DESCRIPTION
**Pull request by sayhaan**
Replace old usages of GDB Index functions to use the new class.

### CODE REVIEW

File: bolt/lib/Rewrite/DWARFRewriter.cpp [hunk 1]
```diff
@@ -435,9 +436,9 @@ static bool getLowAndHighPC(const DIE &Die, const DWARFUnit &DU,
 
 static Expected<llvm::DWARFAddressRangesVector>
 getDIEAddressRanges(const DIE &Die, DWARFUnit &DU) {
-  uint64_t LowPC, HighPC, Index;
-  if (getLowAndHighPC(Die, DU, LowPC, HighPC, Index))
-    return DWARFAddressRangesVector{{LowPC, HighPC, Index}};
+  uint64_t LowPC, HighPC, GDBIndexSection;
```
Comment (ayermolo) [inline bolt/lib/Rewrite/DWARFRewriter.cpp:439]: Was this changed by mistake?

File: bolt/lib/Rewrite/DWARFRewriter.cpp [hunk 2]
```diff
@@ -731,7 +735,8 @@ void DWARFRewriter::updateDebugInfo() {
   auto TheTriple = std::make_unique<Triple>(File->makeTriple());
   std::unique_ptr<DIEStreamer> Streamer =
       createDIEStreamer(*TheTriple, *ObjOS, "TypeStreamer", DIEBlder, *this);
```
Comment (ayermolo) [inline bolt/lib/Rewrite/DWARFRewriter.cpp:737]: Can we pass GDBIndexSection into createDIEStreamer like we do for DIEBuilder so we dont't have to pass it through finalizeTypeSections --> emitUnit?

Comment (sayhaan) [reply bolt/lib/Rewrite/DWARFRewriter.cpp:737]: Wouldn't we still have to pass it through emitUnit? The flow is finalizeTypeSections -> emitUnit -> DIEStreamer::emitUnit -> emitUnitHeader -> emitTypeUnitHeader where GDBIndex::addGDBTypeUnitEntry is called. emitTypeUnitHeader is private so we wouldn't be able to call it from createDIEStreamer.

Comment (ayermolo) [reply bolt/lib/Rewrite/DWARFRewriter.cpp:737]: createDIEStreamer creates DIEStreamer which takes/stores DIEBulldr, etc. So same way pass in GDBIndexSection, so then in emitTypeUnitHeader we can call it like we do DIEBldr

**Review by ayermolo — APPROVED** [review]

**Pull request merged**